lib/efi_loader: fix ABI in efi_mm_communicate_header

etienne-lms opened the pull request:
Pack `struct efi_mm_communicate_header` as done in EDK2. If not the structure takes 4 additional bytes on 32bit targets which breaks ABI.

jenswi-linaro commented on drivers/tee/optee/core.c:
```diff
@@ -302,9 +311,15 @@ static u32 do_call_with_arg(struct udevice *dev, struct optee_msg_arg *arg)
 		struct arm_smccc_res res;
 
+		if (!(gd->flags & GD_FLG_RELOC))
+			flush_shm_dcache(dev, arg);
```
Each time secure world returns the shared memory caches are flushed. So it shouldn't be needed again just before re-entry.

etienne-lms replied:
ok

jenswi-linaro commented on drivers/tee/optee/core.c:
```diff
@@ -292,6 +292,15 @@ static u32 call_err_to_res(u32 call_err)
+static void flush_shm_dcache(struct udevice *dev, struct optee_msg_arg *arg)
+{
+	flush_dcache_range(rounddown((ulong)arg, CONFIG_SYS_CACHELINE_SIZE),
+			   roundup((ulong)arg + sizeof(*arg),
```
You should probably use `OPTEE_MSG_GET_ARG_SIZE(arg->num_params)` instead of `sizeof(*arg)`

etienne-lms replied:
ok, thanks.

etienne-lms committed:
lib/efi_loader: fix ABI in efi_mm_communicate_header

Pack struct efi_mm_communicate_header as done in EDK2 as seen in
release 201808 [1]. If not packed the structure takes 4 additional
bytes on 32bit targets which breaks ABI.

Link: [1] https://github.com/tianocore/edk2/blob/edk2-stable201808/MdePkg/Include/Protocol/MmCommunication.h#L21
Fixes: 23a397d2e2fb ("efi_loader: Add headers for EDK2 StandAloneMM communication")
Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
